app/models/miq_server/worker_management/monitor/kill.rb

```diff
@@ -19,22 +19,16 @@ def remove_unknown_workers
     if bad_workers.size.positive?
       begin
         MiqWorker.inheritance_column = "__disabled"
-        remove_workers(bad_workers)
+        kill_worker_processes(bad_workers)
       ensure
         MiqWorker.inheritance_column = "type"
       end
     end
   end
 
-  def remove_workers(workers = miq_workers)
-    killed_workers = []
-
-    workers.each do |w|
-      w.kill_process if MiqWorker::STATUSES_CURRENT_OR_STARTING.include?(w.status)
-      w.destroy
-      worker_delete(w.pid)
-      killed_workers << w
+  def kill_worker_processes(workers = miq_workers)
+    remove_workers(workers) do |w|
+      w.kill_process if w.current_or_starting?
```

Comment:
This would be simplified if we only wanted to remove the records that were `current_or_starting?`

instead:
-  kill the `killing_or_starting?`
- remove every miq worker from the miq worker list.